terraform/rackspace/modules/etcd_discovery/main.tf

```diff
@@ -0,0 +1,23 @@
+# Variables
```

Comment:
This can be reused, as the null resource isn't tied to any provider. A few minor changes would need to happen in the user data resources in the other providers, but nothing big. My only initial concern with this approach was the fact that the file contents were being passed to the user data template as a var, which would naturally destroy all instances when changing counts, but seeing your issue/comment about the possibility of using a production cluster setup makes this approach acceptable I think.